doc/cabal-package.rst

```diff
@@ -931,34 +939,26 @@ look something like this:
 
     library
         exposed-modules:  Foo.Public
-        build-depends:    foo-internal, base >= 4.3 && < 5
+        build-depends:    foo:foo-internal, base >= 4.3 && < 5
         default-language: Haskell2010
 
     test-suite test-foo
         type:             exitcode-stdio-1.0
         main-is:          test-foo.hs
         -- NOTE: no constraints on 'foo-internal' as same-package
         --       dependencies implicitly refer to the same package instance
-        build-depends:    foo-internal, base
+        build-depends:    foo:foo-internal, base
         default-language: Haskell2010
 
-Sublibraries are also useful for packages that define multiple
-executables, but do not define a publicly accessible library. Internal
-libraries are only visible internally in the package (so they can only
-be added to the :pkg-field:`build-depends` of same-package libraries,
-executables, test suites, etc.) Sublibraries locally shadow any
-packages which have the same name; consequently, don't name an internal
-library with the same name as an external dependency if you need to be
-able to refer to the external dependency in a
-:pkg-field:`build-depends` declaration.
+Another example of the use of internal libraries is a package that includes one
+or more executables but does not include a public library.
 
-Shadowing can be used to vendor an external dependency into a package
-and thus emulate *private dependencies*. Below is an example based on
-a real-world use case:
+Internal libraries can be used to incorporate an external dependency into a
```

Comment:
Again the use of shadowing is key to this example. I don't think it should be cut

Reply:
As above, I don't think shaddowing applies, from Cabal 3.4.

Reply:
This is talking about the file format, not the Cabal version. Note also that currently Hackage refuses versions later than 3.0 on the grounds that they may not be generally usable yet (although I think this is under review), so I'd consider 3.0 the breakpoint for talking about cabal file format versions.

Reply:
My reasoning was that the body of the Guide should discuss the current version of the library and the tool, and the corresponding version of the Cabal specification - because if the Guide was to discuss (at any length) the history of the specification each time it discussed a current feature of the library/tool its complexity could be off-putting to readers and, perhaps, more difficult to maintain. However, if Hackage is de facto limiting the adoption of versions of the specification, I agree it is useful for the Guide to deal with relevant history of the specification back to the point that Hackage accepts. I'll make edits to reflect to the relevant history of the specification of sublibraries in `build-depends:`, and reintroduce 'shadowing' - but not as a 'feature', given that it has subsequently been 'fixed'.